packages/alpinejs/src/directives/x-model.js

```diff
@@ -35,6 +35,17 @@ directive('model', (el, { modifiers, expression }, { effect, cleanup }) => {
 
     cleanup(() => el._x_removeModelListeners['default']())
 
+    // If the input/select/textarea element is linked to a form
+    // we listen for the reset event on the parent form (the event
+    // does not trigger on the single inputs) and update
+    // on nextTick so the page doesn't end up out of sync
+    if (el.form) {
+        let removeResetListener = on(el.form, 'reset', [], (e) => {
+            nextTick(() => el._x_model && el._x_model.set(el.value))
```

Comment:
Is there really risk that `el._x_model` not be defined here?

could `el._x_model?.set(el.value)` not work?

Reply:
I don't think it will ever be undefined but who knows?
I assume you can build a complex case where the reset button has an @click that toggles a boolean, acting on an x-if that removes the input and, because this handler waits for the next tick it may or may not error.

I just thought it was just safer to add a small check. 

the below will work too but I don't have time to check compatibility for all browsers (I'm not a js/frontend dev 😂)

Reply:
Haha true.

Conditional chaining can be transpiled by bundlers when the consumer is targeting older things.